atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/DbAtlasDbFactory.java

```diff
@@ -36,6 +41,14 @@ public KeyValueService createRawKeyValueService(KeyValueServiceConfig config) {
 
     @Override
     public TimestampService createTimestampService(KeyValueService rawKvs) {
-        throw new UnsupportedOperationException("Cannot instantiate a TimestampService from a relational key value service.");
+        Preconditions.checkArgument(rawKvs instanceof DbKvs, "DbAtlasDbFactory expects a raw kvs of type DbKvs, found %s", rawKvs.getClass());
+        DbKvs dbkvs = (DbKvs) rawKvs;
+        return PersistentTimestampService.create(
+                new InDbTimestampBoundStore(
+                        new HikariCPConnectionManager(
```

Comment:
I honestly don't know.  This is a little bit of a mess.

After pushing the connection instantiation into dbkvs it can keep track of the connection pool and expose it here if we want.  In the future we should just rewrite this so it takes a KeyValueService and not a connection pool.